src/visitors/dot_visitor.py

```diff
@@ -111,48 +86,62 @@ def visit_real_variable(self, operator: RealVariable):
+        return self.visit_operator("SUBSTR", operator, 3)
+
+    def visit_string_replacement(self, operator: StringReplacement):
+        return self.visit_operator("STR_RPL", operator, 3)
+
+    def visit_string_variable(self, operator: StringVariable):
+        return self.visit_variable(operator)
 
-    def visitRoot(self, label: str, operator: Operator):
+    def visit_string_literal(self, operator: StringLiteral):
+        return self.visit_constant(operator)
+
+    def visit_string_equality(self, operator: StringEquality):
+        return self.visit_root("STR_EQ", operator)
+
+    def visit_root(self, label: str, operator: Operator):
         heading = "digraph {\n"
         ending = "\n}"
-        id = self.generate_node_name()
-        (op_1, children_1, edges_1) = operator.operator_1.accept(self)
-        (op_2, children_2, edges_2) = operator.operator_2.accept(self)
-        nodes = { id: label, **children_1, **children_2 }
-        edges = { id: [op_1, op_2], **edges_1, **edges_2 }
-        nodes = [ f"    {child} [label={nodes[child]}]" for child in nodes.keys() ]
-        edges = [ (f"    {key} -> {'{' + ' '.join(edges[key]) + '}'}" ) for key in edges.keys() ]
+        name = self.generate_node_name()
+        op_1, children_1, edges_1 = operator.operator_1.accept(self)
+        op_2, children_2, edges_2 = operator.operator_2.accept(self)
+        nodes = {name: label, **children_1, **children_2}
+        edges = {name: [op_1, op_2], **edges_1, **edges_2}
+        nodes = [f"    {child} [label=\"{nodes[child]}\"]" for child in nodes.keys()]
+        edges = [f"    {key} -> {'{' + ' '.join(edges[key]) + '}'}" for key in edges.keys()]
         content = heading + "\n".join(nodes) + "\n" + "\n".join(edges) + ending
         return content
 
-    def visit_binary_operator(self, label: str, operator: Operator):
-        (op_id_1, op_1, edges_1) = operator.operator_1.accept(self)
-        (op_id_2, op_2, edges_2) = operator.operator_2.accept(self)
-        id = self.generate_node_name()
-        return (
-            id,
-            { id: label, **op_1, **op_2},
-            { id: [op_id_1, op_id_2] , **edges_1, **edges_2},
-        )
-
-    def visit_unary_operator(self, label: str, operator: Operator):
-        (op_id_1, op_1, edges_1) = operator.operator_1.accept(self)
-        id = self.generate_node_name()
-        return (
-            id, 
-            { id: label , **op_1 },
-            { id: [op_id_1], **edges_1 }
-        )
-    
+    def visit_operator(self, label: str, operator: Operator, arity: int):
+        op_ids, ops, edges = [], dict(), dict()
+
+        for i in range(arity):
+            op_id, op, sub_edges = getattr(operator, f'operator_{i+1}').accept(self)
+            op_ids.append(op_id)
+            ops = {**ops, **op}
+            edges = {**edges, **sub_edges}
+
+        name = self.generate_node_name()
+        return name, {name: label} | ops, {name: op_ids} | edges,
```

Comment:
I think we missed this one, as the dot visitor is not really used in the tests. For consistency I would still use the other syntax to merge the dictionaries

Reply:
We should probably add at least simple tests for the other visitors at some points. I also just realized that the generation was broken, so we should also add tests for that. Ideally, have the CI pipeline create the operator classes instead of us pushing them if that is possible somehow.

Reply:
I like the idea really much, surely I'll add the step of the generation to the CI, I am not sure about not pushing the files, it can be fine, but I need to think a bit more

Reply:
Ok I will remove the generated files and put them in the gitignore

Reply:
https://github.com/nicdard/fusion-function-generator/pull/45

Reply:
Yep, good point. Overwriting them in the CI pipeline is fine